modules/basics/src/main/java/com/opengamma/strata/basics/schedule/Frequency.java

```diff
@@ -225,10 +252,25 @@ public static Frequency ofWeeks(int weeks) {
    * @throws IllegalArgumentException if months is negative, zero or over 12,000
    */
   public static Frequency ofMonths(int months) {
-    if (months > MAX_MONTHS) {
-      throw new IllegalArgumentException("Months must not exceed 12,000");
+    switch (months) {
+      case 1:
+        return P1M;
+      case 2:
+        return P2M;
+      case 3:
+        return P3M;
+      case 4:
+        return P4M;
+      case 6:
+        return P6M;
+      case 12:
+        return P12M;
+      default:
+        if (months > MAX_MONTHS) {
+          throw new IllegalArgumentException("Months must not exceed 12,000");
```

Comment:
Should use the value of `MAX_MONTHS` in the error message rather than hard-coding to 12,000.